src/python/pants/engine/rules.py

```diff
@@ -21,6 +21,19 @@
 from pants.util.ordered_set import FrozenOrderedSet, OrderedSet
 
 
+class EngineAware(ABC):
```

Comment:
I think `Engine` is a bit more vague/general than intended. How about something like `WorkunitAware`? Based on the docstring, it sounds like this is all about workunits and logging.

Reply:
@Eric-Arellano talking to @asherf yesterday, he preferred a name that referenced the "Engine" rather than "Workunits", on the grounds that people who aren't core pants developers are more likely to understand "sending metadata to the Engine, which is a black box" than the specific concept of Workunits. I'm not sure to what degree we want rule authors to specifically be aware of Workunits as a named concept.

Reply:
I disagree with that logic for the same reasoning you use a month ago. This is a very domain-specific mechanism, so we should use the precise language for that domain, rather than trying to generalize it to make it more accessible.